examples/ros-bag-recorder/adapter/adapter.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/python
+"""
+This file contains the definition of the Adapter itself. The adapter is what allows us to start up the bag recorder
+"""
+
+import logging
+from multiprocessing.sharedctypes import Value
+import threading
+import time
+
+from baghandler import BagHandler
+from config import Config
+from genpy import Duration
+from utils import *
+
+from formant.sdk.agent.v1.client import Client as FormantClient
+
+# logger.basicConfig(level=get_log_level())
+
+logger = logging.getLogger()
+
+
+class Adapter:
+    """
+    The adapter class that will run and control the functionality
+    """
+
+    def __init__(self):
+
+        self.config = Config()
+
+        # Set the topics and types for the specified adapter
+        if self.config.get_param("subscribe_to_all"):
+            self.topics = get_topics()
+        else:
+            self.topics = self.config.get_param("topics")
+
+        # Create a new rosbag with the name as the time and date
+        self.bag_handler = BagHandler()
+        self.bag_thread = threading.Thread(target=self.bag_handler.run, args=(self.is_shutdown,))
+        self.subscriptions = {}
+        self._shutdown_system = False
+        self._recording = False
+        self._recording_thread = None
+        self._fclient = FormantClient() 
+
+        self.bag_thread.start()
+
+    def handle_command(self, message):
+        
+        if message.command == "start_ros_recorder":
+            if self._recording:
+                return
+            
+            self._recording_thread = threading.Thread(target=self.start_recording, args=(None, True))
+            self._recording_thread.start() 
+
+        if message.command == "stop_ros_recorder": 
+            self._recording = False
+            
+        if message.command == "start_ros_recorder_duration":
+            try:
+                duration = rospy.Duration(secs=float(message.text))
+            except ValueError:
+                return
+            self.start_recording(duration, True) 
+
+    def run(self):
+        rospy.init_node("rospy_bag_recorder")
+        rospy.on_shutdown(self.shutdown)
+
+        self._fclient.register_command_request_callback(self.handle_command, 
```

Comment:
Nit: I prefer to just register a different callback for each command so I don't need to bother inspecting `message.command`